### PR TITLE
Fixes #35668 - Show both applicable and installable errata on overview card

### PR DESCRIPTION
**Pull request by jeremylenz**
#### What are the changes introduced in this pull request?

Add a toggle group "Applicable / Installable" (with tooltips) to the errata overview card.
Add an empty state for when a host has unknown errata status.
Update the ErrataTab toggle group with the tooltips as well.

errata-card-new-message
errata-tab-toggle
unknown-errata-status

#### Considerations taken when implementing this change?

We don't currently store applicable errata counts in the DB for each host.  I decided to keep it that way; applicable errata are calculated on demand in the rabl instead of adding new columns to the database.

#### What are the testing steps for this pull request?

1. Get a host with differing numbers of applicable and installable errata
2. View the overview card
3. Verify that when you click on a link, the resulting page is filtered to "All" errata when the toggle group is on "Applicable," and "Installable" when on "Installable."
4. Also view the new tooltips on the Errata page

### CODE REVIEW

File: app/models/katello/host/content_facet.rb [hunk 1]
```diff
@@ -83,13 +83,23 @@ def installable_module_streams(env = nil, content_view = nil)
       end
 
       def errata_counts
-        hash = {
+        installable_hash = {
```
Comment (chris1984) [inline app/models/katello/host/content_facet.rb:86]: Thank you for changing the name on this :)

File: app/models/katello/host/content_facet.rb [hunk 2]
```diff
@@ -83,13 +83,23 @@ def installable_module_streams(env = nil, content_view = nil)
...
-        hash
+        installable_hash[:total] = installable_hash.values.inject(:+)
+        # same for applicable, but we need to get the counts from the db
+        applicable_hash = {
+          :security => applicable_errata.security.count,
+          :bugfix => applicable_errata.bugfix.count,
+          :enhancement => applicable_errata.enhancement.count
+        }
+        applicable_hash[:total] = applicable_hash.values.inject(:+)
+        # keeping installable at the top level for backward compatibility
```
Comment (chris1984) [inline app/models/katello/host/content_facet.rb:99]: :+1:

File: webpack/components/Errata/index.js [hunk 3]
```diff
@@ -178,3 +188,31 @@ export const ErrataSeverity = ({ severity }) => {
...
   severity: PropTypes.string.isRequired,
 };
+
+export const ErrataToggleGroupItem = ({
+  text, tooltipText, isSelected, onChange,
+}) => (
+  <ToggleGroupItem
+    text={
+      <>
+        {text}
```
Comment (chris1984) [inline webpack/components/Errata/index.js:198]: Will this be translated like this instead of using the `_()`

Comment (jeremylenz) [reply webpack/components/Errata/index.js:198]: I used the `__` where it's passed into `<ErrataToggleGroupItem>` so I think that should do it! :slightly_smiling_face:

**Review by chris1984 — APPROVED** [review]
ACK, just had one question.

Code looks fine and tested by doing:
* Registered a host and installed kangaroo latest from the zoo repo
* Downgraded kangaroo and created a filter in the CV to exclude the new package
* Recalcuated the errata so I have installable but not applicable to test how it would handle and looks good:
screenshot-192 168 121 155-2022 10 27-14_22_21

* Changed host back to default org and recalculated errata and it looks good, did not see any console errors either.

**Comment by chris1984** [comment]
+1 to keeping the tool tips question marks

File: webpack/components/extensions/HostDetails/Cards/__tests__/errataOverviewCard.test.js [hunk 4]
```diff
@@ -105,4 +121,72 @@ describe('With errata', () => {
     expect(getByLabelText('20 enhancements')).toBeInTheDocument();
     expect(getByLabelText('10 bug fixes')).toBeInTheDocument();
   });
+
+  test('has cute little tooltips', async () => {
```
Comment (chris1984) [inline webpack/components/extensions/HostDetails/Cards/__tests__/errataOverviewCard.test.js:125]: Love it :heart:

**Comment by jeremylenz** [comment]
I had the link wrong for applicable errata (forgot to change it from `all` to `applicable`. Fixed now (& also wrote a test.)

**Comment by chris1984** [comment]
I will review and test again

File: webpack/components/Errata/errataHelpers.js [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+};
+
+export const friendlyErrataStatus = (errataStatus) => {
+  switch (errataStatus) {
+  case 0:
+    return 'All up to date';
+  case 1:
+    return 'Unknown';
+  // eslint-disable-next-line no-sequences
+  case 2:
```
Comment (chris1984) [inline webpack/components/Errata/errataHelpers.js:25]: Could this be written like this for 2/3

`case (1 | 2) :`

Comment (jeremylenz) [reply webpack/components/Errata/errataHelpers.js:25]:
```js
const a = 2;
=> undefined
const b = 3;
=> undefined
console.log(a | b)
=> 3
```

Bitwise OR is just too confusing. It'd probably work with logical OR `(2 || 3)` but it feels safer to me to stick with the form in the MDN [examples](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/switch).

File: webpack/components/extensions/HostDetails/Cards/ErrataOverviewCard.js [hunk 6]
```diff
@@ -33,74 +38,113 @@ function HostInstallableErrata({
...
+  return (
+    <GridItem rowSpan={1} md={6} lg={4} xl2={3} >
+      <Card ouiaId="errata-card">
+        <CardHeader>
+          <Flex spaceItems={{ default: 'spaceItemsXl' }}>
+            <CardTitle>{__('Errata')}</CardTitle>
+            {neededErrata &&
+              <ToggleGroup isCompact>
+                <ErrataToggleGroupItem
+                  text={__('Applicable')}
```
Comment (chris1984) [inline webpack/components/extensions/HostDetails/Cards/ErrataOverviewCard.js:123]: Should we add `aria-label` to these or just stick with text? Same for below.

**Review by chris1984 — COMMENTED** [review]
Left a few small comments which all but 1 have been addressed. Starting to test it again

File: webpack/components/extensions/HostDetails/Cards/ErrataOverviewCard.js (hunk 6 above)
Comment (jeremylenz) [reply webpack/components/extensions/HostDetails/Cards/ErrataOverviewCard.js:123]: good point! Updated.

File: webpack/components/extensions/HostDetails/Cards/__tests__/errataOverviewCard.test.js (hunk 4 above)
Comment (Andrewgdewar) [reply webpack/components/extensions/HostDetails/Cards/__tests__/errataOverviewCard.test.js:125]: This is fun 😎

**Comment by jeremylenz** [comment]
@MariSvirik Updated to remove question icons (tooltips still activate on hover.)

**Comment by jeremylenz** [comment]
fixed to not show toolbar on happy empty state

**Pull request merged**